### PR TITLE
chore: refactor to wrap client context in BigtableClientContext

### PR DESCRIPTION
**Pull request by mutianf**
Refactor ClientContext creation.

We need to create OpenTelemetry before client context is created so we can inject the PerConnectionErrorTracker interceptor on the ManagedChannel.
We need to access the open telemetry instance later when we create the TracerFactory.

This PR creates a new BigtableCleintContext class that wraps gax ClientContext and OpenTelemetry so we can access both later to avoid creating a global open telemetry instance.

Also moved client context creation logic from EnhancedBigtableStub to BigtableClientContext.

### CODE REVIEW

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java [hunk 1]
```diff
@@ -220,10 +204,11 @@ public BigtableDataClient createForInstance(
             .build();
     ClientContext clientContext =
         sharedClientContext
+            .getClientContext()
             .toBuilder()
             .setTracerFactory(
                 EnhancedBigtableStub.createBigtableTracerFactory(
-                    settings.getStubSettings(), openTelemetry))
+                    settings.getStubSettings(), sharedClientContext.createOpenTelemetry()))
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java:211]: this looks wrong, the otel instance in createOpenTelemetry is tied to thje default resource. This probably a carry over bug, but please add a TODO to fix it

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java [hunk 2]
```diff
@@ -190,10 +173,11 @@ public BigtableDataClient createForInstance(@Nonnull String projectId, @Nonnull
 
     ClientContext clientContext =
         sharedClientContext
+            .getClientContext()
             .toBuilder()
             .setTracerFactory(
                 EnhancedBigtableStub.createBigtableTracerFactory(
-                    settings.getStubSettings(), openTelemetry))
+                    settings.getStubSettings(), sharedClientContext.createOpenTelemetry()))
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java:180]: this looks wrong, the otel instance in createOpenTelemetry is tied to thje default resource. This probably a carry over bug, but please add a TODO to fix it

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java [hunk 3]
```diff
@@ -161,10 +143,11 @@ public BigtableDataClient createForAppProfile(@Nonnull String appProfileId) thro
...
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java:150]: this looks wrong, the otel instance in createOpenTelemetry is tied to thje default resource. This probably a carry over bug, but please add a TODO to fix it

Comment (mutianf) [reply google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java:150]: Sorry, this was a mistake from refactoring the method

File: google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableClientContext.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * This class wraps a gax {@link ClientContext} and an {@link OpenTelemetry} instance that needs to
+ * be created before creating the client context.
```
Comment (igorbernstein2) [inline google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableClientContext.java:45]: Generalize this to: This class wraps all state needed during the life time of the Bigtable client. This includes gax's ClientContext plus and any additional state that Bigtable Client needs

**Review by igorbernstein2 — APPROVED** [review]

**Pull request merged**